os/net/lwip/src/core/ipv4/dhcp.c

```diff
@@ -694,6 +702,28 @@ void dhcp_cleanup(struct netif *netif)
 	}
 }
 
+#if LWIP_NETIF_HOSTNAME
+void dhcp_hostname(struct netif *netif, char *name)
+{
+	char *name_heap;
+
+	if (netif == NULL || name == NULL) {
+		return;
+	}
+
+	if (netif->hostname != NULL) {
+		kmm_free(netif->hostname);
+		netif->hostname = NULL;
+	}
+
+	name_heap = (char *)kmm_malloc(strlen(name) + 1);
+	strncpy(name_heap, name, strlen(name) + 1);
+	name_heap[strlen(name)] = '\0'; //for safety
+
+	netif_set_hostname(netif, name_heap);
```

Comment:
This is what I mentioned before,
'netif_set_hostname' is very simple function but it called by only this function, here, 
Why you separated it? you can assign hostname here.
If netif_set_hostname is common function, it can be expanded in future, then ignore this comment.
I mean If there's chance that it called by another network module, then ignore it.
(But in my opinion, netif_set_hostname is just very simple, it is not necessary actually.)
Also you are setting host name here, then dhcp_set_hostname is acceptable,
dhcp_hostname sounds like constructor of module

Reply:
Thanks for the comment.
Actually, I attached the comment in the previous PR (FYR, https://github.com/Samsung/TizenRT/pull/3681).

Because netif->hostname is not an array but it is char pointer type, and it can be changed on run-time, this malloc/free management should be implemented. And netif_set_hostname is a function provided by open-source LwIP, so we do not want to modify it, considering maintenance.

Thanks for the comment!